pkg/server/grpc_server.go

```diff
@@ -105,6 +109,11 @@ func (g *grpcaCAServer) CreateSigningCertificate(ctx context.Context, request *f
 		if err := csr.CheckSignature(); err != nil {
 			return nil, handleFulcioGRPCError(ctx, codes.InvalidArgument, err, invalidSignature)
 		}
+
+		hashFunc, err = getHashFuncForSignatureAlgorithm(csr.SignatureAlgorithm)
```

Comment:
Unfortunately, this type in the x509 library doesn't expose a way to get information about the hash function (although it's obvious from looking at it). So I've had to write this function that maps from the `crypto.x509.SignatureAlgorithm` to `crypto.Hash` (documented before at https://github.com/golang/go/issues/33317).

Reply:
Shall we get the sign/hash algos pair as part of the `CreateSigningCertificateRequest` parameter? Otherwise, I'm not sure how we can reliably identify which hash algorithm is used by the client. BTW, if I understand it correctly, the hash algorithm used on the client side is probably never really useful on the Fulcio side, except for doing these kind of checks. However we can't infer the hash algo, because you could have (maybe in the future) ECDSA with a different SHA.

Reply:
Yeah, good point. I think we need to include this in `CreateSigningCertificateRequest`. I misunderstood the `SignatureAlgorithm` type as the signing algorithm that the client used not Fulcio itself.

Reply:
We were discussing with @woodruffw that maybe Fulcio should not care about the Hash algorithm used by the client but just about the signing algorithm. Rekor will have to know about the Hash algorithm.

Reply:
Hmm, I think `SignatureAlgorithm` is what we want. In the server, we check the signature by calling [this code](https://github.com/golang/go/blob/master/src/crypto/x509/x509.go#L884) which determines the key and hash algorithm by a similar method.

Reply:
Ok, I think you are right! As the proof of possession in the CSR case is the signature of the CSR itself, using SignatureAlgorithm seems right. 👍